examples/vision_transformer.py

```diff
@@ -0,0 +1,179 @@
+import mygrad as mg
+from mynn.initializers import glorot_normal
+
+
+def gelu(x: mg.Tensor) -> mg.Tensor:
+    """
+    The Gaussian Error Linear Unit described in https://arxiv.org/abs/1606.08415v5 approximated with tanh.
+    """
+    return 0.5 * x * (1 + mg.tanh((2 / mg.pi) ** 0.5 * (x + 0.044715 * x**3)))
+
+
+def layer_norm(x: mg.Tensor, eps: float = 1e-5) -> mg.Tensor:
+    """
+    Layer normalization described in https://arxiv.org/abs/1607.06450v1, with no learnable parameters.
+    """
+    return (x - x.mean(-1, keepdims=True)) / mg.sqrt(x.var(-1, keepdims=True)  + eps)
+
+
+def create_patches(images: mg.Tensor, *, patch_width: int, patch_height: int) -> mg.Tensor:
+    """
+    Create flattened patches of the input image batch.
+
+    Parameters
+        images: mg.Tensor, shape=(B, C, H, W), the batch of images for which to create patches.
+        patch_width: int, the width of the patches to create.
+        patch_height: int, the height of the patches to create.
+
+    Returns
+        mg.Tensor, shape=(B, num_tokens, patch_width * patch_height * C), the patchified input.
+    """
+    b, c, h, w = images.shape
+    h_prime = h // patch_height
+    w_prime = w // patch_width
+    images = images.reshape(b, c, h_prime, patch_height, w_prime, patch_width)
+    images = images.transpose(0, 2, 4, 3, 5, 1)
+    return images.reshape(b, h_prime * w_prime, patch_height * patch_width * c)
```

Comment:
if you don't mind an extra dependency, [einops](https://einops.rocks/) makes this look much cleaner:
```python
return einops.rearrange(
    images, "b c (h p1) (w p2) -> b (h w) (p1 p2 c)", p1=patch_height, p2=patch_width
)
```